src/include/miopen/tensor_reorder_util.hpp

```diff
@@ -0,0 +1,200 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2020-2022 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#ifndef MIOPEN_TENSOR_REORDER_UTIL_HPP_
+#define MIOPEN_TENSOR_REORDER_UTIL_HPP_
+
+#include <miopen/batched_transpose_sol.hpp>
+#include <miopen/general_tensor_reorder_sol.hpp>
+#include <miopen/miopen.h>
+#include <miopen/kernel_info.hpp>
+#include <miopen/op_kernel_args.hpp>
+#include <miopen/execution_context.hpp>
+#include <vector>
+
+namespace miopen {
+struct TensorReorderSolution{
+
+    virtual ~TensorReorderSolution() = default;
+    virtual solver::KernelInfo GetKernel() const = 0;
+    virtual std::vector<OpKernelArg> GetKernelArg() const = 0;
+    virtual std::string GetKernelName() const = 0;
+    virtual bool IsSkippable() const = 0;
+    virtual size_t GetSize() const = 0;
+};
+
+struct WrapperBatchedTransposeSolution_0132 : TensorReorderSolution
```

Comment:
Could you please add some comments to describe what is the 0132, 0231, etc.?

Reply:
Order [0, 1, 3, 2], [0, 2, 3, 1], [0, 3, 1, 2], [2, 3, 0, 1], [3, 0, 1, 2] are using batched transpose kernel to achieve higher performance. Details as following: 
reorder to [0, 1, 3, 2] from [0, 1, 2, 3], we can fix layout index [0] and [1], transpose [2, 3] to [3, 2].
reorder to [0, 2, 3, 1] from [0, 1, 2, 3], we can fix layout index [0], see [2, 3] as an entity, then transpose [1, (2, 3)] to [(2, 3), 1].
reorder to [0, 3, 1, 2] from [0, 1, 2, 3], we can fix layout index [0], see [1, 2] as an entity, then transpose [(1, 2), 3)] to [3, (1, 2)].
reorder to [2, 3, 0, 1] from [0, 1, 2, 3], we can add a fixed layout index , see [0, 1] and [2, 3] as entities, then transpose [(0, 1), (2, 3)] to [(2, 3), (0, 1)].
reorder to [3, 0, 1, 2] from [0, 1, 2, 3], we can add a fixed layout index , see [0, 1, 2] as an entity, then transpose [(0, 1, 2), 3] to [3, (0, 1, 2)].

The reason we have different API like `WrapperBatchedTransposeSolution_0132` is that we choose different fixed index and two dimensions which will be transposed.

Reply:
I mean adding some comments in the code file. Because the other developers might be confused by the series of number.

Reply:
Explanation comments have been added in new commit.